blocks/components/editable/tinymce.js

```diff
@@ -3,7 +3,13 @@ export default class TinyMCE extends wp.element.Component {
 		this.initialize();
 	}
 
-	shouldComponentUpdate() {
+	shouldComponentUpdate( nextProps ) {
+		const isEmpty = String( nextProps.isEmpty );
+
+		if ( this.editorNode.getAttribute( 'data-is-empty' ) !== isEmpty ) {
+			this.editorNode.setAttribute( 'data-is-empty', isEmpty );
+		}
```

Comment:
Maybe. Updating is disabled here, so I'm not sure what the right approach is. @aduth suggested doing this on the wrapper, but that will not have the right styling.